httpcache.go

```diff
@@ -498,6 +506,44 @@ func headerAllCommaSepValues(headers http.Header, name string) []string {
 	return vals
 }
 
+type cachedReadCloser struct {
+	resp     *http.Response
+	origBody io.ReadCloser
```

Comment:
It'll be helpful to document this field. What does original body refer to? Is it the original underlying *http.Response body? Why do we need to keep a reference to it?